src/Stratis.Bitcoin.Tests/NodeConfiguration/NodeSettingsTest.cs

```diff
@@ -148,5 +148,35 @@ public void NodeSettings_CanReadMultiValueFromNone()
             // Assert
             Assert.Empty(result);
         }
+
+        /// <summary>
+        /// Verifies API port value can be passed in on startup.
+        /// </summary>
+        [Fact]
+        public void NodeSettings_CanOverrideOnlyApiPort()
+        {
+            const int apiport = 12345;
+            var nodeSettings = new NodeSettings(networksSelector: Networks.Networks.Bitcoin, args: new[] {  $"-apiport={apiport}" });
+            string result = nodeSettings.ConfigReader.GetOrDefault("apiport", string.Empty);
```

Comment:
OK so you are on the right track here, however you need to now pass `NodeSettings` into `ApiSettings` and then test the overridden value there.  The same for the test below 💪